src/app/ReadClient.cpp

```diff
@@ -1102,6 +1153,12 @@ CHIP_ERROR ReadClient::SendSubscribeRequestImpl(const ReadPrepareParams & aReadP
 
 CHIP_ERROR ReadClient::DefaultResubscribePolicy(CHIP_ERROR aTerminationCause)
 {
+    if (aTerminationCause == CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT)
+    {
+        ChipLogProgress(DataManagement, "ICD device is inactive, skipping scheduling resubscribe within DefaultResubscribePolicy");
+        return CHIP_NO_ERROR;
```

Comment:
So why would we not transition to the InactiveICDSubscription state _here_?  That would make a lot more sense to me...

If we did that, would we still need the special-case in ClearActiveSubscriptionState?  Looks to me like no.

Reply:
Makes sense, however, I think we should set the state right before we calling `mpCallback.OnResubscriptionNeeded` in case the users implemented their own policy.

Reply:
Also, the state is not public (and I don't think we should make it public). the implementation has no way to update the state by their own.